### ISSUE TITLE
Lint should work when directories are passed in

### ISSUE THREAD

**Issue by jgraham** [w3c/web-platform-tests#9879]
`wpt lint` takes a command line option for paths. But it doesn't expand directories, so  e.g. `wpt lint dom` is basically a no-op. To fix this the `lint_paths` function in `tools/lint/lint.py` should check if passed in paths are directories and if so expand the list of paths to include all files rooted on that subtree.

**Comment by teh-f00l** [comment]
I am interested in working on this!

**Comment by jgraham** [comment]
Please go ahead!

**Comment by teh-f00l** [comment]
Update: So I added the files from each directory in the event that a path is specified, but adding a directory argument in the command line is still a no-op. I am thinking this is because I am not able to make use filename strings taken from checking the directory(elements of the paths list are all strings, and when a common path isn't assumed, this can be a problem)

Also when a path is specified is that path the ONLY one we care about, or do we want to work with it's contents in addition to the unexpanded paths? Here is the updated lint_paths code:

```
def lint_paths(kwargs, wpt_root):
   
 if kwargs.get("paths"):
        print(kwargs.get("paths"))
        r = os.path.realpath(wpt_root)
        paths = [os.path.relpath(os.path.realpath(x), r) for x in kwargs["paths"]]
       # this is where my updates start
        for path in paths:
            if os.path.isdir(path):
                if str(path) in kwargs.get("paths"):
                    paths = paths + os.listdir(path)
#my updates end here
    elif kwargs["all"]:
        paths = list(all_filesystem_paths(wpt_root))
    else:
        changed_paths = changed_files(wpt_root)
        force_all = False
        # If we changed the lint itself ensure that we retest everything
        for path in changed_paths:
            path = path.replace(os.path.sep, "/")
            if path == "lint.whitelist" or path.startswith("tools/lint/"):
                force_all = True
                break
        paths = (list(changed_paths) if not force_all
                 else list(all_filesystem_paths(wpt_root)))

    return paths
```

**Comment by foolip** [comment]
The fix for this might be similar to what @domenic asked for in https://github.com/w3c/web-platform-tests/issues/2206. If it fixes that case as well, please close that one too :)

**Comment by teh-f00l** [comment]
will do!

**Comment by teh-f00l** [comment]
I think I resolved the issue, but do we want to allow multiple directories to be passed into the command line? Or is only one enough?

**Comment by domenic** [comment]
For my purposes, a single directory suffices.

**Comment by TimvdLippe** [comment]
This was fixed in #9995 but not closed.

**Comment by foolip** [comment]
Thanks @TimvdLippe!

@domenic, can you see if it works for you, and reopen if not?
